### ISSUE TITLE
Authentication code not working

### ISSUE THREAD

**Issue by SebasTheNinja** [mozilla/fxa#1646]
Hey,
The thing is, every time i want to log in to my firefox account, it says that an auth code has been sent to my emal. I copy the code, paste it but when i click on "continue" it takes me to the previous step (write you email and password).

**Comment by rfk** [comment]
@SebasTheNinja thanks for the report. When this happens:

> i click on "continue" it takes me to the previous step (write you email and password).

Does it give you any error message? Usually this would indicate some sort of authentication problem such as an incorrect password, which I would expect to show an error message.

**Comment by SebasTheNinja** [comment]
No, it won't show any error message. It only sends me to the previous step.

But, somehow, now it sent me a confirmation email instead of the code and
it worked. I don't know

**Comment by clouserw** [comment]
In our triage we're thinking the confirmation codes you received were related to rate limiting for your IP.  Once they expired you got the confirmation link.  I'm going to close this as it appears everything is working now.  Thanks.

**Comment by SebasTheNinja** [comment]
I'm trying to Sync my phone and I'm having the EXACT SAME PROBLEM. This is
so frustrating...

**Comment by SebasTheNinja** [comment]
Is there a way to receive a confirmation email instead of an auth code?

**Comment by shane-tomlinson** [comment]
> Is there a way to receive a confirmation email instead of an auth code?

I'm sorry this is happening @SebasTheNinja. I'm a bit confused what's happening here, you enter in the code sent to your email and it sends you back to the previous step and no error message is displayed?

You can avoid signing codes all together by setting up 2-Factor Authentication(2FA) on the account. Once 2FA is set up, instead of sending emails, we'll ask you to enter a code from your TOTP app.

**Comment by SebasTheNinja** [comment]
Found it. Mozilla was allowing me to log in with an incorrect password so,
when I typed the code, it sent me to the previous step because of the
incorrect password, instead of just showing the "incorrect password" error.

**Comment by shane-tomlinson** [comment]
Thanks for checking again @SebasTheNinja, this is indeed confusing behavior. I thought we had a test for just this case, there was a big debate about the "correct" behavior when the codes were first implemented.

**Comment by shane-tomlinson** [comment]
@SebasTheNinja - You are trying to sign into Sync when this happens, right? We *do* have a test for this case: https://github.com/mozilla/fxa/blob/9e76653ab3bc62e35ebb2487b925ebb9b4cf01cf/packages/fxa-content-server/tests/functional/sign_in_blocked.js#L131. Digging in to see what could be happening here.

**Comment by shane-tomlinson** [comment]
Strange, locally this seems to be working as expected.

<img width="503" alt="Screenshot 2019-07-11 at 11 14 41" src="https://user-images.githubusercontent.com/848085/61043007-2d96ee00-a3cd-11e9-8ba9-0b6ffee5b39f.png">

**Comment by SebasTheNinja** [comment]
No problem!

**Comment by SebasTheNinja** [comment]
Strange. But now the problem is that yesterday i changed my password
(what's when I discovered what was producing the error) and added 2FA, so I
can't provide you a video or something to show you :(

But the way I did it is:
1. Write my email (this one)
2. Try different passwords (I didn't remember mine) until one incorrect
password was accepted.
3. Write the confirmation code sent to email
4. Mozilla sents you to the previous step, sometimes without saying
anything, sometimes showing an "incorrect password" error.
